### PR TITLE
apvs-385 - add AutoApprovalConfig table

### PR DESCRIPTION
**Pull request by pwright08**
apvs-385 - add AutoApprovalConfig table
 
- Migration for AutoApprovalConfig table
- Integration tests for getting and setting config

## Checklist

- [ ] Unit tests
- [ ] Code coverage checked
- [ ] Coding standards
- [ ] Error Handling
- [ ] Security/performance considered?
- [ ] Deployment changes considered?
- [ ] README updated

### CODE REVIEW

File: migrations/20161205161306_add_auto_approval_config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    table.string('Caseworker', 100)
+    table.dateTime('DateCreated')
+    table.boolean('AutoApprovalEnabled')
+    table.decimal('CostVariancePercentage')
+    table.decimal('MaxClaimTotal')
+    table.integer('MaxDaysAfterAPVUVisit')
+    table.integer('MaxNumberOfClaimsPerYear')
+    table.string('RulesDisabled', 200)
+    table.boolean('IsEnabled')
+  })
```
Comment (stevenalexander) [inline migrations/20161205161306_add_auto_approval_config.js]: after creating table add default config based on config defaults so we don't have to manually use the internal screen on all envs

File: test/integration/services/data/test-get-auto-approval-config.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return knex('AutoApprovalConfig')
+      .insert([{
+        Caseworker: 'caseworker1@test.com',
+        DateCreated: moment().toDate(),
+        AutoApprovalEnabled: 'true',
+        CostVariancePercentage: '5.00',
+        MaxClaimTotal: '100.00',
+        MaxDaysAfterAPVUVisit: '28',
+        MaxNumberOfClaimsPerYear: '10',
+        RulesDisabled: 'auto-approval-rule-1, auto-approval-rule-2, auto-approval-rule-3',
```
Comment (stevenalexander) [inline test/integration/services/data/test-get-auto-approval-config.js]: no spaces in delimit

File: app/services/data/update-auto-approval-config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const config = require('../../../knexfile').intweb
+const knex = require('knex')(config)
+const moment = require('moment')
+
+module.exports = function (caseworker, autoApprovalEnabled, costVariancePercentage, maxClaimTotal,
+  maxDaysAfterAPVUVisit, maxNumberOfClaimsPerYear, rulesDisabled) {
```
Comment (stevenalexander) [inline app/services/data/update-auto-approval-config.js]: rulesDisabled should be an array of check names, build delimited list from this

File: migrations/20161205161306_add_auto_approval_config.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  .then(function () {
+    return knex('AutoApprovalConfig')
+      .insert({
+        Caseworker: null,
+        DateCreated: moment().toDate(),
+        AutoApprovalEnabled: config.AUTO_APPROVAL_ENABLED,
+        CostVariancePercentage: config.AUTO_APPROVAL_COST_VARIANCE_PERCENTAGE,
+        MaxClaimTotal: config.AUTO_APPROVAL_MAX_CLAIM_TOTAL,
+        MaxDaysAfterAPVUVisit: config.AUTO_APPROVAL_MAX_DAYS_AFTER_APVU_VISIT,
+        MaxNumberOfClaimsPerYear: config.AUTO_APPROVAL_MAX_NUMBER_OF_CLAIMS_PER_YEAR,
```
Comment (stevenalexander) [inline migrations/20161205161306_add_auto_approval_config.js]: use static values so we can be sure what values will be inserted no matter what time the migration runs

File: config.js [hunk 5]
```diff
@@ -38,5 +38,12 @@ module.exports = {
...
   // Value configurations
-  MAX_APPROVED_DIFFERENT_AMOUNT: process.env.MAX_APPROVED_DIFFERENT_AMOUNT || '250'
+  MAX_APPROVED_DIFFERENT_AMOUNT: process.env.MAX_APPROVED_DIFFERENT_AMOUNT || '250',
+
+  // Auto approval default configuration
+  AUTO_APPROVAL_ENABLED: process.env.APVS_AUTO_APPROVAL_ENABLED || 'true',
+  AUTO_APPROVAL_COST_VARIANCE_PERCENTAGE: process.env.APVS_AUTO_APPROVAL_COST_VARIANCE_PERCENTAGE || '0.1',
+  AUTO_APPROVAL_MAX_CLAIM_TOTAL: process.env.APVS_AUTO_APPROVAL_MAX_CLAIM_TOTAL || '250',
+  AUTO_APPROVAL_MAX_DAYS_AFTER_APVU_VISIT: process.env.APVS_AUTO_APPROVAL_MAX_DAYS_AFTER_APVU_VISIT || '28',
+  AUTO_APPROVAL_MAX_NUMBER_OF_CLAIMS_PER_YEAR: process.env.APVS_AUTO_APPROVAL_MAX_NUMBER_OF_CLAIMS_PER_YEAR || '26'
```
Comment (stevenalexander) [inline config.js]: remove these

**Comment by pwright08** [comment]
closing until related internal screen is ready

File: app/services/data/get-auto-approval-config.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const config = require('../../../knexfile').intweb
+const knex = require('knex')(config)
+
+module.exports = function () {
+  return knex('AutoApprovalConfig')
+    .where('IsEnabled', true)
+    .orderBy('DateCreated', 'desc')
+    .first()
```
Comment (stevenalexander) [inline app/services/data/get-auto-approval-config.js]: split rulesDisabled here for output

**Pull request closed without merging**